docs/migration/cosmoshub-2.md

```diff
@@ -0,0 +1,208 @@
+software version and genesis file hash before starting your validator and signing.
+
+The riskiest thing a validator can do is discover that they made a mistake and repeat the upgrade
+procedure again during the network startup. If you discover a mistake in the process, the best thing
+to do is wait for the network to start before correcting it. If the network is halted and you have
+started with a different genesis file than the expected one, seek advice from a Tendermint developer
+before resetting your validator.
+
+## Recovery
+
+Prior to exporting `cosmoshub-2` state, validators are encouraged to take a full data snapshot at the
+export height before proceeding. Snapshotting depends heavily on infrastructure, but generally this
+can be done by backing up the `.gaiacli` and `.gaiad` directories.
+
+In the event that the upgrade does not succeed, validators and operators must downgrade back to
+v0.34.6+ of the _Cosmos SDK_ and restore to their latest snapshot before restarting their nodes.
+
+## Upgrade Procedure
+
+__Note__: It is assumed you are currently operating a full-node running v0.34.6+ of the _Cosmos SDK_.
+
+- The version/commit hash of Gaia v2.0.3: `2f6783e298f25ff4e12cb84549777053ab88749a`
+- The upgrade height as agreed upon by governance: **[PLACEHOLDER]**
+- You may obtain the canonical UTC timestamp of the exported block by any of the following methods:
+  - Block explorer (e.g. [Hubble](https://hubble.figment.network/cosmos/chains/cosmoshub-2/blocks/[PLACEHOLDER]?format=json&kind=block))
+  - Through manually querying an RPC node (e.g. `/block?height=[PLACEHOLDER]`)
+  - Through manually querying a Gaia REST client (e.g. `/blocks/[PLACEHOLDER]`)
+
+1. Verify you are currently running the correct version (v0.34.6+) of the _Cosmos SDK_:
+
+```shell
+$ gaiad version --long
+cosmos-sdk: 0.34.6
+git commit: 80234baf91a15dd9a7df8dca38677b66b8d148c1
+vendor hash: f60176672270c09455c01e9d880079ba36130df4f5cd89df58b6701f50b13aad
+build tags: netgo ledger
+go version go1.12.2 linux/amd64
+```
+
+2. Export existing state from `cosmoshub-2`
+
+**NOTE**: It is recommended for validators and operators to take a full data snapshot at the export
+height before proceeding in case the upgrade does not go as planned or if not enough voting power
+comes online in a sufficient and agreed upon amount of time. In such a case, the chain will fallback
+to continue operating `cosmoshub-2`. See [Recovery](#recovery) for details on how to proceed.
+
+```shell
+$ gaiad export --for-zero-height --height=[PLACEHOLDER] > cosmoshub_2_genesis_export.json
+```
+
+3. Verify the SHA256 of the (sorted) exported genesis file:
+
+```shell
+$ jq -S -c -M '' cosmoshub_2_genesis_export.json | shasum -a 256
+[PLACEHOLDER]  cosmoshub_2_genesis_export.json
+```
+
+4. At this point you now have a valid exported genesis state! All further steps now require
+v2.0.3 of [Gaia](https://github.com/cosmos/gaia).
+
```

Comment:
We should add a note about golang version, check that is 1.13.x or upgrade

Reply:
updated.